tasks/sep/032-op-pectra-defence/VALIDATION.md

```diff
@@ -0,0 +1,126 @@
+# Validation
+
+This document can be used to validate the state diff resulting from the execution of the upgrade transaction.
+
+For each contract listed in the state diff, please verify that no contracts or state changes shown in the Tenderly diff
+are missing from this document. Additionally, please verify that for each contract:
+
+- The following state changes (and none others) are made to that contract. This validates that no unexpected state
+  changes occur.
+- All addresses (in section headers and storage values) match the provided name, using the Etherscan and Superchain
+  Registry links provided. This validates the bytecode deployed at the addresses contains the correct logic.
+- All key values match the semantic meaning provided, which can be validated using the storage layout links provided.
+
+## Nested Safe State Overrides and Changes
+
+This task is executed by the nested 2/2 `ProxyAdminOwner` Safe. Refer to the
+[generic nested Safe execution validation document](../../../NESTED-VALIDATION.md)
+for the expected state overrides and changes.
+
+The `approvedHashes` mapping **key** of the `ProxyAdminOwner` that should change during the simulation is
+- Council simulation: `0x3394b41f3c05f4c1bb2767a59f18fd9cc4497d7a8781dca9206396044c5d4a80`
+- Foundation simulation: `0xfd5f9ae7ec744b3e9bbfd555849eb2f5727c46fe3e5cd532c7e102c0934e99c1`
```

Comment:
Is this right? I only checked foundation so far, but the key (slot) should be `0x1329022094b502dadea5bcf711a39060bcad420131d2d543625c040c8d842d56`, which I get from:

```
cast index bytes32 $SAFE_HASH $(cast index address $SAFE_SIGNER 8)
```

where `SAFE_HASH=0x940b6861a5af2179064ee1ed84f388c84988ee03a2312e3fc9b610d125925fa7` and `SAFE_SIGNER=0xDEe57160aAfCF04c34C887B5962D0a69676d3C8B`. This matches with what I see in tenderly

Reply:
Ok, and yea SC one should be `0x9b0798be757313db24bf52fe1666fe7f2be5416c0c6e5ea626f281f4afc14641` from my sim, `cast index bytes32 0x940b6861a5af2179064ee1ed84f388c84988ee03a2312e3fc9b610d125925fa7 $(cast index address 0xf64bc17485f0B4Ea5F06A96514182FC4cB561977 8)`

Reply:
Made that change.